### DISCUSSION TITLE
Migrating Gutenberg Blocks Plugin to Tailwind CSS v4: How to Enable User-Defined Tailwind Classes?

### DISCUSSION THREAD

**Discussion by NafisMahmudAyon** [tailwindlabs/tailwindcss#15281] [Help]
I’m working on a WordPress Gutenberg blocks plugin where users can add custom Tailwind CSS classes for styling the blocks directly.

Previously, I was using Tailwind CSS v3.4.15 with a CDN to generate and apply the CSS dynamically. Now, I want to migrate to Tailwind CSS v4.

With the new features and changes in Tailwind CSS v4 (like @layer, updated border defaults, and other optimizations), how should I adapt this process?

- How can I continue to allow users to dynamically add Tailwind classes within the Gutenberg editor?
- Is there a way to use Tailwind CSS v4 with a CDN for runtime styling, or do I need to transition to a build process?
- Any best practices for integrating Tailwind v4 into WordPress plugins while keeping the workflow flexible for end users?

Would appreciate any guidance or tips from those who’ve made a similar migration!

**Comment by Vince-ALIEN** [comment]
Hi @NafisMahmudAyon,

Whta do you think of that ?

https://github.com/Vince-ALIEN/ufo-blocks
